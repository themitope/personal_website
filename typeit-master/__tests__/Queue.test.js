import Queue from "../src/Queue.js";

let queue;

beforeEach(() => {
  queue = new Queue(["first-action", null, {}]);
});

describe("add()", () => {
  test("It should add steps properly.", () => {
    queue.add(["step1"]);
    queue.add(["step2"]);

    expect(queue.getItems()).toMatchSnapshot();
  });

  test("It should add a step passed as a function.", () => {
    const myFunc = () => "func!";
    queue.add(myFunc);
    queue.add(myFunc);
    queue.add(myFunc);
    expect(queue.getItems()).toMatchSnapshot();
  });

  test("It should add multiple steps passed at once.", () => {
    const myFunc = () => "func!";
    queue.add([
      [myFunc, 1],
      [myFunc, 2],
      [myFunc, 3]
    ]);
    expect(queue.getItems()).toMatchSnapshot();
  });
});

test("It should reset properly by marking each item as having not yet been executed.", () => {
  queue.add([
    ["value1", null, { executed: true }],
    ["value2", null, { executed: true }],
    ["value3", null, { executed: false }],
    ["value4", null, { executed: true }],
    ["value5", null, { executed: true }]
  ]);

  queue.reset();

  expect(queue.getItems()).toMatchSnapshot();
});

test("It should set initial steps properly.", () => {
  let items = [
    ["value1", null, {}],
    ["value2", null, {}],
    ["value3", null, {}]
  ];

  let q1 = new Queue(items);

  expect(q1.getItems()).toMatchSnapshot();
});

test("It should only return non-executed items.", () => {
  let items = [
    ["value1", null, { executed: true }],
    ["value2", null, { executed: false }],
    ["value3", null, {}]
  ];

  let q1 = new Queue(items);

  expect(q1.getItems()).toMatchSnapshot();
});

test("It should return no items if all are executed.", () => {
  let items = [
    ["value1", null, { executed: true }],
    ["value2", null, { executed: true }],
    ["value3", null, { executed: true }]
  ];

  let q1 = new Queue(items);

  expect(q1.getItems()).toEqual([]);
});
