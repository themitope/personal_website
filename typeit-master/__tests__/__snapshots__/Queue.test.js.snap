// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should only return non-executed items. 1`] = `
Array [
  Array [
    "value2",
    null,
    Object {
      "executed": false,
      "id": 1,
    },
  ],
  Array [
    "value3",
    null,
    Object {
      "id": 2,
    },
  ],
]
`;

exports[`It should reset properly by marking each item as having not yet been executed. 1`] = `
Array [
  Array [
    "first-action",
    null,
    Object {
      "executed": false,
      "id": 0,
    },
  ],
  Array [
    "value1",
    null,
    Object {
      "executed": false,
      "id": 1,
    },
  ],
  Array [
    "value2",
    null,
    Object {
      "executed": false,
      "id": 2,
    },
  ],
  Array [
    "value3",
    null,
    Object {
      "executed": false,
      "id": 3,
    },
  ],
  Array [
    "value4",
    null,
    Object {
      "executed": false,
      "id": 4,
    },
  ],
  Array [
    "value5",
    null,
    Object {
      "executed": false,
      "id": 5,
    },
  ],
]
`;

exports[`It should set initial steps properly. 1`] = `
Array [
  Array [
    "value1",
    null,
    Object {
      "id": 0,
    },
  ],
  Array [
    "value2",
    null,
    Object {
      "id": 1,
    },
  ],
  Array [
    "value3",
    null,
    Object {
      "id": 2,
    },
  ],
]
`;

exports[`add() It should add a step passed as a function. 1`] = `
Array [
  Array [
    "first-action",
    null,
    Object {
      "id": 0,
    },
  ],
  Array [
    [Function],
    null,
    Object {
      "id": 1,
    },
  ],
  Array [
    [Function],
    null,
    Object {
      "id": 2,
    },
  ],
  Array [
    [Function],
    null,
    Object {
      "id": 3,
    },
  ],
]
`;

exports[`add() It should add multiple steps passed at once. 1`] = `
Array [
  Array [
    "first-action",
    null,
    Object {
      "id": 0,
    },
  ],
  Array [
    [Function],
    1,
    Object {
      "id": 1,
    },
  ],
  Array [
    [Function],
    2,
    Object {
      "id": 2,
    },
  ],
  Array [
    [Function],
    3,
    Object {
      "id": 3,
    },
  ],
]
`;

exports[`add() It should add steps properly. 1`] = `
Array [
  Array [
    "first-action",
    null,
    Object {
      "id": 0,
    },
  ],
  Array [
    "step1",
    null,
    Object {
      "id": 1,
    },
  ],
  Array [
    "step2",
    null,
    Object {
      "id": 2,
    },
  ],
]
`;
